hbase-server/src/main/java/org/apache/hadoop/hbase/io/hfile/LruBlockCache.java

```diff
@@ -592,10 +593,27 @@ public int evictBlocksByHfileName(String hfileName) {
    * @return the heap size of evicted block
    */
   protected long evictBlock(LruCachedBlock block, boolean evictedByEvictionProcess) {
-    LruCachedBlock previous = map.remove(block.getCacheKey());
-    if (previous == null) {
+    // We need something effectively final for the lambda.
+    final AtomicBoolean evicted = new AtomicBoolean(false);
+    map.computeIfPresent(block.getCacheKey(), (k, v) -> {
```

Comment:
Check the declaration, the map is a ConcurrentHashMap so it is fine. Better add some comments to say that we must use ConcurrentHashMap here, as for CSLM, the computeIfPresent method is not atomic, which means the supplier could be called multiple times.

Reply:
Let me also bullet proof the lambda against multiple invocation.